pkg/services/sqlstore/migrations/encrypt_passwords.go

```diff
@@ -0,0 +1,208 @@
+package migrations
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"github.com/go-xorm/xorm"
+	. "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"hash"
+	"io"
+)
+
+type AddEncryptPasswordsMigration struct {
+	MigrationBase
+}
+
+func (m *AddEncryptPasswordsMigration) Sql(dialect Dialect) string {
+	return "code migration"
+}
+
+func (m *AddEncryptPasswordsMigration) Exec(sess *xorm.Session, mg *Migrator) error {
+	var passwordRows []map[string]string
+	datasourceTypes := []string{
```

Comment:
These seems to be the only ones that use plaintext passwords, if there are other please let me know.